otelcol/flags.go

```diff
@@ -51,7 +51,7 @@ func flags(reg *featuregate.Registry) *flag.FlagSet {
 		})
 
 	flagSet.Var(featuregate.NewFlag(reg), featureGatesFlag,
-		"Comma-delimited list of feature gate identifiers. Prefix with '-' to disable the feature. '+' or no prefix will enable the feature.")
+		"Comma-delimited list of feature gate identifiers. Prefix with '-' to disable the feature. '+' or no prefix will enable the feature. Start with strict to enable strict mode.")
```

Comment:
Sure thing - please note as a positional argument it must be placed before feature gates are set though. I can do that or maybe I can use a `-feature-gates-strict` argument that would enforce strictness. `-feature-gates` and `-feature-gates-strict` would be mutually exclusive. Would that work for you?

Reply:
Going with that and let's see how it lands.

Reply:
> please note as a positional argument it must be placed before feature gates are set though

@atoulme could you elaborate on this? I don't understand why

Reply:
It’s my guess, but the flags are applied sequentially. Strict mode applies when the flag for feature gates is set.

Reply:
I don't think the fact that flags are applied sequentially means we can't make this work in any order. We can either 
1. delay setting flags until all flags have been parsed (e.g. by using a `PreRun` hook from cobra to actually apply the feature gates)
2. apply feature gates if `--feature-gates` is passed before the strict mode flag, but fail if we don't pass strict mode validation

I guess we should do (2) if people have been complaining that feature gate initialization happens to late as in #4967. The only case where this would make a difference is if there is a feature gate that changes something about strict mode or CLI flag parsing.

Reply:
1. seems like it will clash with #4967 then.
2. seems ok with me and I can do it.
What do you think of my current approach of using `--feature-gates-strict`?